app/services/katello/pulp/content_counts_calculator.rb

```diff
@@ -0,0 +1,52 @@
+module Katello
+  module Pulp
+    class ContentCountsCalculator
+
+      def initialize(repos)
+        @repos = repos
+      end
+
+      def calculate
+        counts = {
+          :packages => 0,
+          :puppet_modules => 0,
+          :yum_repositories => 0,
+          :docker_repositories => 0,
+          :puppet_repositories => 0,
+          :errata => 0
+        }
+
+        @repos.each do |r|
+          if is_repo_type?(r, 'rpm')
+            counts[:yum_repositories] += 1
+            counts[:packages] += get_unit_count(r, 'rpm')
+            counts[:errata] += get_unit_count(r, 'erratum')
+          elsif is_repo_type?(r, 'docker')
+            counts[:docker_repositories] += 1
+            counts[:docker_images] += get_unit_count(r, 'docker_image')
+          elsif is_repo_type?(r, 'puppet')
+            counts[:puppet_repositories] += 1
+            counts[:puppet_modules] += get_unit_count(r, 'puppet_module')
+          end
+        end
+
+        counts
+      end
+
+      protected
+
+      def is_repo_type?(repo, repo_type)
+        repo['notes'] && (repo['notes']['_repo-type'] == "#{repo_type}-repo")
+      end
+
+      def get_unit_count(repo, unit_type)
+        if repo['content_unit_counts']
+          repo['content_unit_counts'][unit_type] || 0
```

Comment:
0 is imho more suitable here because we calculate sum for a content view across repos of multiple types